src/test/java/com/example/resilience4jdemo/TestServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package com.example.resilience4jdemo;
+
+import io.github.resilience4j.bulkhead.BulkheadFullException;
+import io.github.resilience4j.circuitbreaker.CallNotPermittedException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import com.example.resilience4jdemo.helpers.HttpClient;
+import com.example.resilience4jdemo.service.TestService;
+
+import reactor.core.publisher.Mono;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class TestServiceTest {
+
+    @Mock
+    private HttpClient httpClient;
+
+    @InjectMocks
+    private TestService testService;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void testFastResponse() {
+        when(HttpClient.genericGet(any(String.class), any())).thenReturn(Mono.just("Fast response"));
+        String response = testService.fastResponse();
+        assertEquals("Fast response", response);
```

Comment:
**Suggestion:** Consider using `verify` to ensure that the `HttpClient.genericGet` method is called with the expected parameters. This will make your tests more robust and prevent bugs due to incorrect parameters. [best practice]
```suggestion
    @Test
    public void testFastResponse() {
        when(HttpClient.genericGet(any(String.class), any())).thenReturn(Mono.just("Fast response"));
        String response = testService.fastResponse();
        assertEquals("Fast response", response);
        verify(httpClient).genericGet(any(String.class), any());
    }
```